packages/remix-dev/cli/create.ts

```diff
@@ -433,24 +544,66 @@ type TemplateType =
   // local directory
   | "local";
 
-async function detectTemplateType(
-  template: string,
-  useTypeScript: boolean,
-  token?: string
-): Promise<TemplateType> {
-  if (template.startsWith("file://") || fse.existsSync(template)) {
-    return "local";
+function detectTemplateType(template: string): TemplateType {
+  // 1. Check if the user passed a local file. If they hand us an explicit file
+  //    URL, we'll validate it first. Otherwise we just ping the filesystem to
+  //    see if the string references a filepath and, if not, move on.
```

Comment:
This ordered flow makes sense to me - nice work 👍

Reply:
Agreed 💯